packages/sit-onyx/src/components/OnyxSelectDialog/OnyxSelectDialog.vue

```diff
@@ -0,0 +1,162 @@
+<script lang="ts" setup generic="TValue extends string">
+import { ref, watchEffect } from "vue";
+import { injectI18n } from "../../i18n";
+import OnyxButton from "../OnyxButton/OnyxButton.vue";
+import OnyxCard from "../OnyxCard/OnyxCard.vue";
+import OnyxHeadline from "../OnyxHeadline/OnyxHeadline.vue";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import OnyxModalDialog from "../OnyxModalDialog/OnyxModalDialog.vue";
+import OnyxVisuallyHidden from "../OnyxVisuallyHidden/OnyxVisuallyHidden.vue";
+import type { OnyxSelectDialogProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxSelectDialogProps<TValue>>(), {
+  open: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the color scheme should be changed.
+   */
+  "update:modelValue": [value: TValue];
+  /**
+   * Emitted when the dialog should be closed.
+   */
+  close: [];
+}>();
+
+const slots = defineSlots<{
+  /**
+   * Optional slot to add custom content, e.g. a description to the dialog header (below the headline).
+   */
+  description?(): unknown;
+}>();
+
+const currentValue = ref<TValue>();
+watchEffect(() => (currentValue.value = props.modelValue));
+
+const { t } = injectI18n();
+
+const handleChange = (event: Event) => {
+  const target = event.target as HTMLInputElement;
+  currentValue.value = target.value as TValue;
+};
+
+const handleApply = () => {
+  if (!currentValue.value) return;
+  emit("update:modelValue", currentValue.value);
+  emit("close");
+};
+</script>
+
+<template>
+  <OnyxModalDialog
+    v-bind="props"
+    class="onyx-select-dialog"
+    :label="props.label"
+    @close="emit('close')"
+  >
+    <template v-if="!!slots.description" #description>
+      <slot name="description"></slot>
+    </template>
+
+    <form class="onyx-select-dialog__form" @submit.prevent="handleApply">
+      <fieldset class="onyx-select-dialog__list" @change="handleChange">
+        <label v-for="option in props.options" :key="option.value">
+          <OnyxCard class="onyx-select-dialog__option">
+            <OnyxIcon v-if="option.icon" :icon="option.icon" class="onyx-select-dialog__icon" />
+
+            <div>
+              <OnyxVisuallyHidden is="div">
+                <input
+                  type="radio"
+                  name="color-scheme"
+                  :autofocus="props.modelValue === option.value"
```

Comment:
It would be handy to have a autofocus even when nothing is selected or require the modelValue prop:
```suggestion
                  :autofocus="props.modelValue === option.value || (!props.modelValue && index === 0)"
```

Reply:
I made the modelValue required since such dialogs (color scheme, application language etc.) will always have a value defined